erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -656,10 +641,13 @@ def get_items(self):
 							item["to_warehouse"] = self.pro_doc.wip_warehouse
 					self.add_to_stock_entry_detail(item_dict)
 
-				elif self.work_order and (self.purpose == "Manufacture" or self.purpose == "Material Consumption for Manufacture") and \
+				elif self.work_order and self.purpose == "Manufacture" and \
 					frappe.db.get_single_value("Manufacturing Settings", "backflush_raw_materials_based_on")== "Material Transferred for Manufacture":
 					self.get_transfered_raw_materials()
 
+				elif self.work_order and self.purpose == "Material Consumption for Manufacture":
+					self.get_transfered_raw_materials()
+
```

Comment:
It is not right. If "backflush_raw_materials_based_on" is based on BOM, then it should not consider transferred materials. It will simply assume all the items in the BOM is in WIP warehouses.